devito/data/allocators.py

```diff
@@ -306,6 +307,55 @@ def put_local(self):
         return self._node == 'local'
 
 
+class ExternalAllocator(MemoryAllocator):
+
+    """
+    External allocator for definig Function symbols with data loaded by the
+    user, without copying it to Devito's allocated memory.
```

Comment:
the point is: devito doesn't allocate any memory

So, how about

```
An ExternalAllocator is used to assign pre-existing user data to Functions. Thus, Devito does not allocate any memory. 
```

Reply:
updated